common-auth/src/main/java/com/hortonworks/registries/auth/server/KerberosBasicAuthenticationHandler.java

```diff
@@ -0,0 +1,215 @@
+        final UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(identity, password);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Created authentication token for principal {} with name {} and is authenticated {}", token.getPrincipal(),
+                    token.getName(), token.isAuthenticated());
+        }
+
+        Authentication authentication = null;
+
+        try {
+            authentication = provider.authenticate(token);
+        } catch (BadCredentialsException ex) {
+            LOG.debug("Bad credentials provided", ex);
+        } catch (final org.springframework.security.core.AuthenticationException ex) {
+            LOG.error("Kerberos login failed.", ex);
+        }
+
+        if (authentication == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return null;
+        } else {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Ran provider.authenticate() and returned authentication for principal {} with name {} and is authenticated {}",
+                        authentication.getPrincipal(), authentication.getName(), authentication.isAuthenticated());
+            }
+            response.setStatus(HttpServletResponse.SC_OK);
+            return new AuthenticationToken(kerberosName.getShortName(), authentication.getName(), getType());
+        }
+    }
+
+    /**
+     * Get the kerberos user principal.
+     */
+    private String getUserIdentity(KerberosName kerberosName, String rawPrincipal) {
+        String defaultRealm = kerberosName.getDefaultRealm();
+        // The default realm configured from the krb5 configuration will end up being used
+        boolean realmInRawPrincipal = !Utils.isBlank(kerberosName.getRealm());
+        final String identity;
+        if (realmInRawPrincipal) {
+            // there's a realm already in the given principal, use it
+            identity = rawPrincipal;
+            LOG.debug("Realm was specified in principal {}, default realm was not added to the identity being authenticated", rawPrincipal);
+        } else if (!Utils.isBlank(defaultRealm)) {
+            // the value for the default realm is not blank, append the realm to the given principal
+            identity = rawPrincipal + "@" + defaultRealm;
+            LOG.debug("Realm was not specified in principal {}, default realm {} was added to the identity being authenticated", rawPrincipal,
+                    defaultRealm);
+        } else {
+            // otherwise, use the given principal, which will use the default realm as specified in the krb5 configuration
+            identity = rawPrincipal;
+            LOG.debug("Realm was not specified in principal {}, default realm is blank and was not added to the identity being authenticated",
+                    rawPrincipal);
+        }
+
+        return identity;
+    }
+
+    static class KerberosUserDetailsService implements UserDetailsService {
```

Comment:
Do we need static keyword here?

Reply:
static nested classes are usually preferred if the class is not using any of the fields from the enclosing class.

Reply:
Agreed. I hurried up. Did not pay much attention. Considering no access to enclosing class, static should have been finer for this case!